src/ranged.cpp

```diff
@@ -1556,11 +1556,13 @@ static void cycle_action( item &weap, const tripoint &pos )
         cargo = vp->vehicle().get_parts_at( pos, "CARGO", part_status_flag::any );
     }
 
+    item *brass_catcher = weap.gunmod_find( itype_brass_catcher );
```

Comment:
use a flag instead of a specific itype.

Reply:
So give a flag to brass catcher and make a gunmod_find alternative that looks for flags?